### DISCUSSION TITLE
Selected Option

### DISCUSSION THREAD

**Discussion by dominikgeimer** [wireui/wireui#377] [Q&A] [answered]
Hi,

important things first: This package is just awesome! Thanks for that @PH7-Jack! 

Can someone give me a hint how to set the selected option of a select element?

I am using the spatie/laravel-permission package. I have a Modal where I can edit a user and change the role of the user. But the user already has a role (for example, 'admin'). 

How can I use a select (for example, this one: https://livewire-wireui.com/docs/select#option-with-description) and have the current role already selected?

I hope you guys understand what I mean.

Thanks
Dominik

**Comment by ph7jack** [comment, marked as answer]
Hello @dominikgeimer 
Thanks for appreciating the WireUI

To have the selected option outside a livewire component, just pass the value attribute into the select with the value from the database
```
<x-select
    label="User Role"
    placeholder="Select one role"
    :options="[
        ['name' => 'Admin',    'value' => 'admin',   'description' => 'Some description here'],
        ['name' => 'Manager',  'value' => 'manager', 'description' => 'Some description here'],
        ['name' => 'Seller',   'value' => 'seller',  'description' => 'Some description here'],
        ['name' => 'User',     'value' => 'user',    'description' => 'Some description here'],
    ]"
    option-label="name"
    option-value="value"
    value="manager"
/>
```
If your select is inside a livewire component, just set the **wire:model** and set the value to the model property
```
<x-select
    label="User Role"
    placeholder="Select one role"
    :options="[
        ['name' => 'Admin',    'value' => 'admin',   'description' => 'Some description here'],
        ['name' => 'Manager',  'value' => 'manager', 'description' => 'Some description here'],
        ['name' => 'Seller',   'value' => 'seller',  'description' => 'Some description here'],
        ['name' => 'User',     'value' => 'user',    'description' => 'Some description here'],
    ]"
    option-label="name"
    option-value="value"
    wire:model="model"
/>
```

**Reply by jonathanoeijoeng** [reply]
Can it be done in multi select? I did it but got error The {option-value} and {option-label} attributes cannot be used with primitive options values: string, integer, double, boolean, NULL 
![image](https://user-images.githubusercontent.com/7046560/185726203-418bead6-3d70-442a-8d89-b3ecca1c6a0d.png)
![image](https://user-images.githubusercontent.com/7046560/185726212-fc342666-b2bb-4135-8c3a-f18acaff9a2c.png)

**Reply by ph7jack** [reply]
@jonathanoeijoeng Yes, is the same way using the multiselect.
Just make sure your value/model is an array
If you are having doubts, open a new issue/discussion
